File: haskell solutions/problem_3.hs
--returning the biggest prime factor of a number


factors:: Int -> [Int]
factors n = filter (\x -> rem n x == 0) [1..round((fromIntegral n)/2)]

prime_factors:: [Int] -> [Int]
prime_factors myList = filter (\x -> is_prime x == True) myList

is_prime:: Int -> Bool
is_prime n = if length (filter (\x -> rem n x == 0) [1..round(sqrt(fromIntegral n))]) == 1 
    then True 
    else False

main:: IO()
main = do
    let factors_list = factors 5143
    let prime_factors_list = prime_factors factors_list
    let biggest_prime = last prime_factors_list
    print biggest_prime

File: python solutions/Problems 1-10/problem_3.py
from math import sqrt, floor


def factors(n):
    myList = []
    for i in range(1, floor(n / 2)):
        if n % i == 0:
            prime = True
            for j in range(2, floor(sqrt(i))):
                print(i, j)
                if i % j == 0:
                    prime = False
                    break
            if prime == True:
                myList.append(i)

    return myList[-1]


result = factors(5143)
print("The biggest prime factor is: ", result)
